### ISSUE TITLE
New Rule: validateNewlineInModuleImports? (ES6)

### ISSUE THREAD

**Issue by hzoo** [jscs-dev/node-jscs#1304]
What do you think about a similar rule to `validateNewlineAfterArrayElements` (Issue #317, commited as https://github.com/jscs-dev/node-jscs/commit/71fd2bbf6524382aeb980eb5f02366e62ec9f9fc) but for ES6 module imports? There is also already a similar request for objects #1115.

```js
const { a } = require('./abc');
const { cube, diag, square } = require('./lib');
import { cube, diag, square } from 'lib';
```

```js
const { a } = require('./abc');
const {
  cube,
  diag,
  square
} = require('./lib');

import {
  cube,
  diag,
  square
} from 'lib';
```

**Comment by zxqfox** [comment]
These are destructors in Modules Imports and Consts. Probably this rule can be more common.

**Comment by hzoo** [comment]
Yeah I know it seemed like it to me but from [the comments at the bottom](http://odetocode.com/blogs/scott/archive/2014/09/11/features-of-es6-part-6-destructuring.aspx).

"In ES6 modules, you don’t technically use destructuring: the syntax is similar, but you are importing bindings, not values." and the AST is different as well. So I don't know if we want the same rule or different rules for naming purposes. I guess we might want 2 different rules?

**Comment by hzoo** [comment]
We released [3.0](https://github.com/jscs-dev/node-jscs/releases/tag/v3.0.0) recently! We also wrote a [post](https://medium.com/@markelog/jscs-end-of-the-line-bc9bf0b3fdb2#.oot3wvgpu) about how 3.0 will be our last major release (since we are deprecating the project) and that we are [merging with ESLint](http://eslint.org/blog/2016/04/welcoming-jscs-to-eslint)!

Thus we won't be working on any new features moving forward which includes new rules/options. We will continue to fix bugs and work on CST/better autofixing for current rules in the upcoming months both to help JSCS users and to gain experience that we can hopefully use in ESLint. But otherwise we both teams will be working on getting [ESLint feature parity with JSCS](https://github.com/eslint/eslint/milestones/JSCS%20Compatibility) and just a nice transition for our users.

I'm putting an `orphaned` label to note this issue isn't planned to be worked on, but might be a good contribution as a custom rule (not in JSCS core itself).

I would encourage anyone who wants the rule to try writing it themselves: I would either write a custom rule or submit an issue to ESLint and if accepted a PR. Otherwise a custom rule that won't be in core repo itself.

JSCS rules are also very similar to ESLint rules: here are some resources to get started: 

- [Project-specific Lint Rules with ESLint](https://gist.github.com/jareware/7179093)
- [Creating Custom Rules for ESLint](https://insideops.wordpress.com/2015/12/08/creating-custom-rules-for-eslint)
- [Maintaining JavaScript Code Quality with ESLint ](https://www.paypal-engineering.com/2014/12/12/maintaining-javascript-code-quality-with-eslint/)
